common/test/acceptance/pages/lms/instructor_dashboard.py

```diff
@@ -659,16 +668,43 @@ def is_cohorted(self, state):
             self.q(css=self._bounded_selector('.cohorts-state')).first.click()
             self.wait_for_ajax()
 
-    def toggles_showing_of_discussion_topics(self):
+    def cohort_management_controls_visible(self):
+        """
+        Return the visibility status of cohort management controls(cohort selector section etc).
+        """
+        return (self.q(css=self._bounded_selector('.cohort-management-nav')).visible and
+                self.q(css=self._bounded_selector('.wrapper-cohort-supplemental')).visible)
+
+
+class DiscussionManagementSection(PageObject):
+
+    url = None
+
+    discussion_form_selectors = {
+        'course-wide': '.cohort-course-wide-discussions-form',
+        'inline': '.cohort-inline-discussions-form'
+    }
+
+    def is_browser_on_page(self):
+        return self.q(css=self.discussion_form_selectors['course-wide']).present
+
+    def _bounded_selector(self, selector):
+        """
+        Return `selector`, but limited to the divided discussion management context.
+        """
+        return '.discussions-management {}'.format(selector)
+
+    def is_save_button_disabled(self, key):
         """
-        Shows the discussion topics.
+        Returns the status for form's save button, enabled or disabled.
         """
-        self.q(css=self._bounded_selector(".toggle-cohort-management-discussions")).first.click()
-        self.wait_for_element_visibility("#cohort-discussions-management", "Waiting for discussions to appear")
+        save_button_css = '%s %s' % (self.discussion_form_selectors[key], '.action-save')
+        disabled = self.q(css=self._bounded_selector(save_button_css)).attrs('disabled')
+        return disabled[0] == 'true'
 
     def discussion_topics_visible(self):
         """
-        Returns the visibility status of cohort discussion controls.
+        Returns the visibility status of divide discussion controls.
         """
         EmptyPromise(
```

Comment:
I'm not sure we need this promise now-- isn't the check in is_browser_on_page sufficient? I'm wondering if this is leftover from when you had to click the link to show the discussion tree.